### PR TITLE
[SPARK-23602][SQL] PrintToStderr prints value also in interpreted mode

### PR DESCRIPTION
**Pull request by mgaido91**
## What changes were proposed in this pull request?

`PrintToStderr` was doing what is it supposed to only when code generation is enabled.
The PR adds the same behavior in interpreted mode too. 

## How was this patch tested?

added UT

### CODE REVIEW

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/MiscExpressionsSuite.scala [hunk 1]
```diff
@@ -43,4 +45,21 @@ class MiscExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     checkEvaluation(Length(Uuid()), 36)
     assert(evaluateWithoutCodegen(Uuid()) !== evaluateWithoutCodegen(Uuid()))
   }
+
+  test("PrintToStderr") {
+    val errorStream = new java.io.ByteArrayOutputStream()
+    val systemErr = System.err
```
Comment (hvanhovell) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/MiscExpressionsSuite.scala]: Can you wrap setting the `stderr` in a `try { ... } finally { ... }`?

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/MiscExpressionsSuite.scala [hunk 2]
```diff
@@ -47,18 +47,24 @@ class MiscExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
...
+
+    val (outputEval, outputCodegen) = try {
+      val errorStream = new java.io.ByteArrayOutputStream()
+      System.setErr(new PrintStream(errorStream))
+      // check without codegen
+      checkEvaluationWithoutCodegen(PrintToStderr(inputExpr), 1)
+      val outputEval = errorStream.toString
+      errorStream.reset()
+      // check with codegen
+      checkEvaluationWithoutCodegen(PrintToStderr(inputExpr), 1)
```
Comment (hvanhovell) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/MiscExpressionsSuite.scala]: `checkEvaluationWithoutCodegen`?  Shouldn't we use `checkEvaluationWithUnsafeProjection` or `checkEvaluationWithGeneratedMutableProjection`?

Comment (mgaido91) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/MiscExpressionsSuite.scala]: yes, sorry, my bad

**Review by hvanhovell — APPROVED** [review]
LGTM - pending jenkins.

**Comment by hvanhovell** [comment]
retest this please

**Comment by hvanhovell** [comment]
Merging to master. Thanks!

**Pull request closed without merging**